### PR TITLE
Add TimedCache and use it to cache config values

### PR DESCRIPTION
**Pull request by Azquelt**
Use it to cache:
 - server.xml app properties
 - raw values retrieved from any source

For #11079

### CODE REVIEW

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
Most recently started build will pick up the changes which were pushed just after it was requested.

File: dev/com.ibm.ws.microprofile.config.1.4/src/com/ibm/ws/microprofile/config14/impl/Config14Impl.java [hunk 1]
```diff
@@ -47,13 +50,14 @@
      */
     public Config14Impl(ConversionManager conversionManager, SortedSources sources, ScheduledExecutorService executor, long refreshInterval) {
         super(conversionManager, sources);
+        rawValueCache = new TimedCache<>(executor, 500, TimeUnit.MILLISECONDS);
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.config.1.4/src/com/ibm/ws/microprofile/config14/impl/Config14Impl.java]: The 500 should be in a constant somewhere, possibly a constants file and possibly configurable??

File: dev/com.ibm.ws.microprofile.config.1.4/src/com/ibm/ws/microprofile/config14/impl/Config14Impl.java [hunk 2]
```diff
@@ -135,6 +144,18 @@ private SourcedValue getRawValue(String key) {
         return raw;
     }
 
+    @Override
+    public void close() {
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.config.1.4/src/com/ibm/ws/microprofile/config14/impl/Config14Impl.java]: Does the `convertedValueCache` need clearing?

Comment (Azquelt) [reply dev/com.ibm.ws.microprofile.config.1.4/src/com/ibm/ws/microprofile/config14/impl/Config14Impl.java]: No. Potentially that could be an improvement, but this whole class should be GC'd shortly after `close()` is called.

The only reason for closing `TimedCache` is to cancel any pending invalidation task.

File: dev/com.ibm.ws.microprofile.config.1.4/src/com/ibm/ws/microprofile/config14/impl/Config14Impl.java (hunk 1 above)
Comment (Azquelt) [reply dev/com.ibm.ws.microprofile.config.1.4/src/com/ibm/ws/microprofile/config14/impl/Config14Impl.java]: Unless you object, I'll do that as a follow up.

File: dev/com.ibm.ws.microprofile.config.1.4/test/src/com/ibm/ws/microprofile/config14/test/ConversionTest.java [hunk 3]
```diff
@@ -47,14 +47,16 @@ public void testBadCharUnicode() {
...
-    public void testConverterCache() {
+    public void testConverterCache() throws InterruptedException {
         Config config = ConfigProviderResolver.instance().getBuilder().addDefaultSources().withConverter(TestType.class, 100, new TestTypeConverter()).build();
         String key = "testConverterCache";
 
         String rawString = "This is the RAW String 1";
         String expected = rawString + " - 1";
         System.setProperty(key, rawString);
 
+        Thread.sleep(700); // Let the raw value cache expire
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.config.1.4/test/src/com/ibm/ws/microprofile/config14/test/ConversionTest.java]: should be based on the `500` constant ... unless that is made configurable

**Comment by tevans78** [comment]
#run-libby-bot

**Comment by tevans78** [comment]
LibbyBot has fallen asleep?! Merging anyway.

**Pull request merged**